src/arrayMethodJoin.js

```diff
@@ -1,12 +1,30 @@
 'use strict';
 
-/**
+/*
  * Implement method join
  *
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function() {
-    // write code here
+  [].__proto__.join2 = function(valueBetweenElements = ',') {
+    const elementBetweenItem = valueBetweenElements + '';
+    const lengthOfArray = this.length;
+    let str = '';
+    if (!lengthOfArray) {
+      return str;
+    }
+    for (let i = 0; i < lengthOfArray; i++) {
+      const itemOfArray = this[i];
+      if (itemOfArray || itemOfArray === false ||
+        isNaN(itemOfArray) || itemOfArray === 0) {
```

Comment:
Why do you need so complex condition?
Also, put all expression to one line or each to new line